java/client/src/org/openqa/selenium/support/pagefactory/ByChained.java

```diff
@@ -48,10 +48,21 @@ public ByChained(By... bys) {
 
   @Override
   public WebElement findElement(SearchContext context) {
-    List<WebElement> elements = findElements(context);
-    if (elements.isEmpty())
-      throw new NoSuchElementException("Cannot locate an element using " + toString());
-    return elements.get(0);
+    if (bys.length == 0) {
+      throw new NoSuchElementException("No Bys were specified in this " + getClass().getSimpleName());
+    }
+
+    WebElement elem = null;
+    for (By by : bys) {
+      boolean isFirstIteration = elem == null;
+      if (isFirstIteration) {
+        elem = by.findElement(context);
+      } else {
+        elem = elem.findElement(by);
+      }
+    }
+    assert elem != null; // NoSuchElement should have already been thrown
```

Comment:
The question now is whether using `WebElement#findElement` instead `findElements` is really an improvement.
Because originally I quite blindly assumed that retrieving one element must be more efficient than retrieving more, but actually I don't know..

So let's analyse..
I think we can say that most of the times `WebElement#findElement` will be resolved as `RemoteWebElement#findElement`, right?
So, by following through the calls, the question ends up being whether 
`Command.FIND_CHILD_ELEMENT` and `DriverCommand.FIND_ELEMENT` are more efficient than `DriverCommand.FIND_CHILD_ELEMENTS` and `DriverCommand.FIND_ELEMENTS` (respectively).